src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs

```diff
@@ -134,6 +140,27 @@ public List<VotingData> GetAndCleanScheduledVotes()
             }
         }
 
+        /// <summary>Checks pending polls against finished polls and removes pending polls that will make no difference and basically are redundant.</summary>
+        /// <remarks>All access should be protected by <see cref="locker"/>.</remarks>
+        private void CleanFinishedPollsLocked()
+        {
+            // We take polls that are not pending (collected enough votes in favor) but not executed yet (maxReorg blocks
+            // didn't pass since the vote that made the poll pass). We can't just take not pending polls because of the
+            // following scenario: federation adds a hash or fed member or does any other revertable action, then reverts
+            // the action (removes the hash) and then reapplies it again. To allow for this scenario we have to exclude
+            // executed polls here.
+            List<Poll> finishedPolls = this.polls.Where(x => !x.IsPending && !x.IsExecuted).ToList();
+
+            for (int i = this.scheduledVotingData.Count - 1; i >= 0; i--)
```

Comment:
Any reason, why you are executing this `for` backwards?

Reply:
Yeah, we are removing items in the list we are queering. Another way would be to do for forward but collect all the items for removal into a new collection and then do the removal, but that approach is slower.